benchkit/hdc/__init__.py

```diff
@@ -0,0 +1,220 @@
+# SPDX-License-Identifier: MIT
+"""
+Module to handle hdc (OpenHarmony Device Connector) interactions between host and remote phone.
+HDC documentation: https://docs.openharmony.cn/pages/v5.0/en/application-dev/dfx/hdc.md 
+"""
+from typing import Iterable, Optional, Callable
+from enum import Enum
+
+from benchkit.shell.shell import get_args, shell_out
+from benchkit.utils.types import Command, PathType
+
+
+class HDCError(Exception):
+    """Handle for errors from hdc."""
+
+class DeviceIdentifierKind(Enum):
+    IP_AND_PORT = 0,
+    SERIAL = 1,
+    DONT_CARE = 2,
+
+class HDCDevice:
+    """Representation of a device connected through hdc."""
+
+    def __init__(
+        self,
+        identifier: str,
+        kind: DeviceIdentifierKind = DeviceIdentifierKind.DONT_CARE,
```

Comment:
instead of having identified and kind, it might be nicer to create to sub-classes of HDCDevice: HDCLocalDevice, and HDCRemoteDevice. They would basically only have a different constructor, and perhaps `__str__` implementation and everything else would be implemented in the superclass.

Reply:
are we implementing this in this PR? @TriedAngle

Reply:
> instead of having identified and kind, it might be nicer to create to sub-classes of HDCDevice: HDCLocalDevice, and HDCRemoteDevice. They would basically only have a different constructor, and perhaps `__str__` implementation and everything else would be implemented in the superclass.

What is the difference between a Local and Remote HDC device? Aren't both reachable by serial or ip:port ?

Reply:
Currently we don't really need to differentiate between device kinds, since when calling `hdc` we don't need to care about if the identifier is a serial number or an IP. 
I would be fine with completely removing `kind`, and just taking an arbitrary identifier str as input.